pydantic/utils.py

```diff
@@ -104,7 +105,8 @@ def validate_field_name(bases: List[Type['BaseModel']], field_name: str) -> None
 
 
 def lenient_issubclass(cls: Any, class_or_tuple: Union[Type[Any], Tuple[Type[Any], ...]]) -> bool:
-    return isinstance(cls, type) and issubclass(cls, class_or_tuple)
+    """Fallback for `typing.Pattern` as it is not a valid class"""
+    return isinstance(cls, type) and issubclass(cls, class_or_tuple) or cls is class_or_tuple is Pattern
```

Comment:
We may have a better check later to support for example `Pattern[str]` but `typing` API is changing so much so we'll probably first need some utils to retrieve the right informations